### PR TITLE
Immigrate test gson scenario.

### PR DESCRIPTION
**Pull request by wayilau**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance
- [x] Tests cases

- Related issues
#3583
___
Move an optional plugin gson tests cases to main repo.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Ref ##3583

**Comment by wu-sheng** [comment]
Your YAML format breaks again :) Please fix it. And I think GSON doesn't require a new group. Add it to `PluginsTest / Webflux_SocketIO_RabbitMQ`, and rename the task please.

**Comment by wayilau** [comment]
> Your YAML format breaks again :) Please fix it. And I think GSON doesn't require a new group. Add it to `PluginsTest / Webflux_SocketIO_RabbitMQ`, and rename the task please.

OK.

File: .github/workflows/plugins-test.yaml [hunk 1]
```diff
@@ -118,6 +118,8 @@ jobs:
         run: bash test/plugin/run.sh netty-socketio-scenario
       - name: Run rabbitmq-scenario (12)
         run: bash test/plugin/run.sh rabbitmq-scenario
+      - name Run gson-scenario (6)
```
Comment (wu-sheng) [inline .github/workflows/plugins-test.yaml]: Your format is still not right. You need to pay more attention to details :)

Comment (wayilau) [reply .github/workflows/plugins-test.yaml]: i am sorry. i've checked.

File: .github/workflows/plugins-test.yaml [hunk 2]
```diff
@@ -118,6 +118,8 @@ jobs:
         run: bash test/plugin/run.sh netty-socketio-scenario
       - name: Run rabbitmq-scenario (12)
         run: bash test/plugin/run.sh rabbitmq-scenario
+      - name: Run gson-scenario (6)
+        run: bash test/plugin/run.sh gson-scenario
```
Comment (wu-sheng) [inline .github/workflows/plugins-test.yaml]: You add this to the group, but you haven't change the group name.

Comment (wayilau) [reply .github/workflows/plugins-test.yaml]: You  mean i change the name to 'Webflux_SocketIO_GSON', am i right ?

Comment (wu-sheng) [reply .github/workflows/plugins-test.yaml]: Yes, this name. But should be `Toolkit_Webflux_SocketIO_RabbitMQ_GSON`

Comment (wayilau) [reply .github/workflows/plugins-test.yaml]: OK, Got it.

File: test/plugin/scenarios/gson-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+registryItems:
+  applications:
+    - {gson-scenario: nq 0}
+  instances:
+    - {gson-scenario: nq 0}
+  operationNames:
+    - gson-scenario: [/gson-scenario/case/healthCheck,/gson-scenario/case/gson-scenario]
```
Comment (wu-sheng) [inline test/plugin/scenarios/gson-scenario/config/expectedData.yaml]: healthCheck should not be included in expected data today.

File: test/plugin/scenarios/gson-scenario/support-version.list [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# lists your version here
+
+2.8.5
```
Comment (wu-sheng) [inline test/plugin/scenarios/gson-scenario/support-version.list]: 2.8.6 has released, is it working? If so, please add.

File: .github/workflows/plugins-test.yaml [hunk 5]
```diff
@@ -118,6 +118,8 @@ jobs:
         run: bash test/plugin/run.sh netty-socketio-scenario
       - name: Run rabbitmq-scenario (12)
         run: bash test/plugin/run.sh rabbitmq-scenario
+      - name: Run gson-scenario (6)
```
Comment (wu-sheng) [inline .github/workflows/plugins-test.yaml]: If 2.8.6 supported, this should be changed too.

Comment (wayilau) [reply .github/workflows/plugins-test.yaml]: Got it .

File: test/plugin/scenarios/gson-scenario/config/expectedData.yaml (hunk 3 above)
Comment (wayilau) [reply test/plugin/scenarios/gson-scenario/config/expectedData.yaml]: I will remove.

**Pull request merged**